wagtail_localize/views/update_translations.py

```diff
@@ -153,28 +179,17 @@ def post(self, request, **kwargs):
 
     @transaction.atomic
     def form_valid(self, form):
-        self.object.update_from_db()
-
         enabled_translations = self.object.translations.filter(enabled=True)
-        if form.cleaned_data.get("use_machine_translation"):
-            machine_translator = get_machine_translator()
-            for translation in enabled_translations.select_related("target_locale"):
-                apply_machine_translation(
-                    translation.id, self.request.user, machine_translator
-                )
 
-        if form.cleaned_data["publish_translations"]:
-            for translation in enabled_translations.select_related("target_locale"):
-                with contextlib.suppress(ValidationError):
-                    translation.save_target(user=self.request.user, publish=True)
-        else:
-            for translation in enabled_translations.select_related(
-                "source", "target_locale"
-            ):
-                with contextlib.suppress(ValidationError):
-                    translation.source.update_target_view_restrictions(
-                        translation.target_locale
-                    )
+        update_translations(
+            self.request,
+            self.object,
+            enabled_translations,
+            use_machine_translation=form.cleaned_data.get(
+                "use_machine_translation", False
+            ),
+            publish_translations=form.cleaned_data["publish_translations"],
+        )
 
         self.components.save(
```

Comment:
With publish+auto-machine-translate we cannot really handle those components, I think. With bulk actions we may add those components to the confirmation view. That's why I have not extracted them into `update_translations()`.

I also don't fully understand them. Are those components meant to be created once per translation request or once per translation source? I know, it's up to the dev, but it would be nice to get some infos on how this has been used so far.